### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'find_next'

### ISSUE THREAD

**Issue by Kintoru** [mkayeterry/usda-fdc-data#3]
Kinto@DarkFantasy MINGW64 ~
$ python3 /c/Users/Kinto/Downloads/usda-fdc-data-main/main.py

Initializing processing of USDA FDC data. Output directory set to:
> fdc_data

Traceback (most recent call last):
  File "C:\Users\Kinto\Downloads\usda-fdc-data-main\main.py", line 61, in <module>
    usda_urls = get_usda_urls()
  File "C:\Users\Kinto\Downloads\usda-fdc-data-main\preprocessing\_utils.py", line 26, in get_usda_urls
    'h2', text='Latest Downloads').find_next('table')
                                   ^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'find_next'

**Comment by mkayeterry** [comment]
Yeah I see this, will look into it further today but will have to make some changes. 

['https://fdc.nal.usda.gov/fdc-datasets/FoodData_Central_foundation_food_csv_2024-10-31.zip', 'https://fdc.nal.usda.gov/fdc-datasets/FoodData_Central_sr_legacy_food_csv_2018-04.zip', 'https://fdc.nal.usda.gov/fdc-datasets/FoodData_Central_branded_food_csv_2024-10-31.zip']

You should be able to use these links to bypass get_usda_urls in the meantime

**Comment by Kintoru** [comment]
Thank you! I've been messing with the code all night but have had no luck.

**Comment by mkayeterry** [comment]
There was a slight change in the newest datasets they published, so I'm assuming you didn't make it past processing foundation. I fixed the issue and will push up shortly - just running through one final check. I'll likely have to make some more updates in the near future so that it can keep pulling in the latest datasets, but should be good to go for now.

**Comment by mkayeterry** [comment]
I'm going to close this for now, but please let me know if you're still having trouble!

**Comment by Kintoru** [comment]
Thank you so much! It is working as intended now!

Foundation did give me some trouble. At first, when I posted the AttributeError, I realized the html.parser couldn't get past the app-root tag so I tried to switch it to lxml and html5lib but for whatever reason Git Bash was not identifying that I had both of those parsers installed so it kept giving me an error about the treebuilder not being found, so I had to transfer the html5lib installation into the project folder and I got rid of the lxml code since it was having further issues with not being able to import ETREE from lxml, I got it to download finally after tweaking the URLs for the new names. Then you posted the initial comment I had issues with foundation_dir for a while so I traced back everything, then all_dir was the issue, so messed with that. Got it to work with switching the column names to look for inside food_attribtue.csv and then it got stuck with an error due to something else which I assume had to do with how the data is processed afterwards but I was too sleepy after looking at it for hours so I just knocked out haha. Thank you again!
